src/mavlink-router/endpoint.cpp

```diff
@@ -238,6 +239,7 @@ int Endpoint::read_msg(struct buffer *pbuf, int *target_sysid, int *target_compi
             msg_entry = nullptr;
         } else {
             _add_sys_comp_id(((uint16_t)*src_sysid << 8) | *src_compid);
+            *crc_valid = true;
```

Comment:
Can  you make sure it's always set:
```
*crc_valid = _check_crc(msg_entry);
if (*crc_valid) {
...
```

Reply:
Yes because it's only used when the return is `>0`. The same for e.g. `src_sysid`.
If you wanted to be more cautious you could also just initialize it at the entry of the method, which is fine too.
But not setting it to false is asking for trouble as `read_msg` is called in a loop.

Reply:
Aaah you're right. Don't know how I missed the aspect about the loop :facepalm: 
Thanks!